synthesizer/program/src/function/mod.rs

```diff
@@ -69,6 +69,11 @@ impl<N: Network, Instruction: InstructionTrait<N>, Command: CommandTrait<N>> Fun
         self.inputs.iter().map(|input| input.value_type()).cloned().collect()
     }
 
+    /// Returns the function input type variants.
+    pub fn input_variants(&self) -> Vec<Variant> {
```

Comment:
nit: these could return an `impl Iterator<Item = Variant>`

Reply:
I left it consistent with the existing `output_types` and `input_types` functions. We can revisit in the future for possible optimizations.